### DISCUSSION TITLE
Jasonelle Web

### DISCUSSION THREAD

**Discussion by cbornet** [jasonelle-archive/jasonelle-v2#104] [Ideas]
It would be great to have support for WEB browsers.
Maybe forking Jasonette-Web and taking from where it was left ?

**Comment by clsource** [comment]
Jasonette-Web was a beta prototyping tool using cell.js
reviving that tool would mean one of these options:

1 - to support cell.js again
2 - or to change the tool using another web framework

I found that creating a whole engine for the web that interprets the jasonette json structures as a prototyping tool
would be a huge time commitment since it would be necessary to keep it updated with the latest versions.

In my opinion is best to use a tool such as Jason APK for prototyping and testing since it would be using the same engine as the core and the mobile environment.

**Comment by cbornet** [comment]
Having Jasonelle-Web would allow to design only once for both desktop and mobile which would be a huge win!

**Comment by clsource** [comment]
While I do find the use of a web version of Jasonette feasible and useful.
Jasonelle's main focus is on mobile Android and iOS.

However if there is such a project that focused on the web part. We would happily link them and redirect users that want to create web apps using that tech. 👍

**Comment by Nicero** [comment]
Jasonette / Jasonelle is the most intelligent approach to mobile app development ever seen. As said by Ethan itself, one of the main purpose of Jasonette is to let developers convert their ideas into apps in the fastest way possible. For this reason I think that having a web editor as an IDE where the developer draws and tests its app is a fundamental part of the project. I tried many times Jasonette and I came back to it often still convinced by its potential, but having to wait for the JSON to be loaded to the device every single little change I make to the code, makes you lose one of the main advantages of Jasonette.
I really hope that in the near future Jasonette-web will be resumed.

**Comment by clsource** [comment]
> Jasonette / Jasonelle is the most intelligent approach to mobile app development ever seen. As said by Ethan itself, one of the main purpose of Jasonette is to let developers convert their ideas into apps in the fastest way possible. For this reason I think that having a web editor as an IDE where the developer draws and tests its app is a fundamental part of the project. I tried many times Jasonette and I came back to it often still convinced by its potential, but having to wait for the JSON to be loaded to the device every single little change I make to the code, makes you lose one of the main advantages of Jasonette.
> I really hope that in the near future Jasonette-web will be resumed.

@Nicero Yes improving the developer UX is one of the keypoints of the project. Having a tool that helps development by using drag and drop and seeing your results in nearly realtime would be wonderful. But that kind of DX can also be achieved by websockets connection and a paired simulator/device that would render the output on change. There is no need to rebuilt a new tool from scratch if we could use what we have today. A web interface can be considered, but I don't know about the web renderer part.

**Comment by Nicero** [comment]
Glad to know this possibility. Could you explain how to do it? The only way I worked with Jasonette up to now is loading the Janosette-Android APK in Android Studio, link to the JSON and the run the APK, but as you know, in this way the APK is built every time and this approach is not feasible. Maybe a dedicated post/tutorial on this argument would be apreciated by many other users.

**Comment by clsource** [comment]
The quickest way for now is using Jason app. An APK That you can use to prototype using an URL. 

https://github.com/jasonelle/jasonette-android/releases/download/v2.1/Jasonelle-jason.apk
https://github.com/jasonelle-archive/docs/tree/develop/examples/jasonette/apps/jason-app


The websocket part needs more thinkering an experimentation. But the integration can be done using the builing blocks https://github.com/Jasonette/agent.websockets

**Comment by eazuka** [comment]
Sorry if my question seem off. i'm a bit confuse. 
Is Jasonelle Web same as the Jasonette JSON Web Container and is the JSON Web Container feature not longer path of Jasonelle?

Final question, is Webview in Jasonelle same as JSON Web Container?

**Reply by clsource** [reply]
Jasonette Web was a project made by Ethan (the original author of jasonette) that intended to create something similar to Jasonette in a Web environment.

JSON Webcointainer is just a component inside Jasonette so you can load websites in your app :)
